spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/MicrometerTracingAutoConfiguration.java

```diff
@@ -70,20 +69,20 @@ public DefaultTracingObservationHandler defaultTracingObservationHandler(Tracer
 
 	@Bean
 	@ConditionalOnMissingBean
-	@ConditionalOnBean({ Tracer.class, HttpServerHandler.class })
-	@Order(HTTP_SERVER_TRACING_OBSERVATION_HANDLER_ORDER)
-	public HttpServerTracingObservationHandler httpServerTracingObservationHandler(Tracer tracer,
-			HttpServerHandler httpServerHandler) {
-		return new HttpServerTracingObservationHandler(tracer, httpServerHandler);
+	@ConditionalOnBean({ Tracer.class, Propagator.class })
+	@Order(RECEIVER_TRACING_OBSERVATION_HANDLER_ORDER)
+	public PropagatingReceiverTracingObservationHandler<?> propagatingReceiverTracingObservationHandler(Tracer tracer,
```

Comment:
I merged https://github.com/spring-projects/spring-boot/pull/31783 into this PR but there might be an issue around these beans. If a `Propagator` is present both `PropagatingReceiverTracingObservationHandler` and `PropagatingSenderTracingObservationHandler` will be created. See my comment on https://github.com/spring-projects/spring-boot/pull/31783.

/cc @marcingrzejszczak

Reply:
Is this actually a problem that both beans would be then created if the `Propagator` bean is there?

Reply:
It's just an extra bean or two that might not be needed.